torch/onnx/_type_utils.py

```diff
@@ -108,36 +113,86 @@ def from_name(
     @classmethod
     @_beartype.beartype
     def from_dtype(cls, dtype: torch.dtype) -> JitScalarType:
-        """Convert a torch dtype to ScalarType."""
+        """Convert a torch dtype to JitScalarType.
+
+        NB: DO NOT USE this API if dtype is comes from a torch._C.Value.type().dtype() call because
+            "RuntimeError: INTERNAL ASSERT FAILED at "../aten/src/ATen/core/jit_type_base.h" can
```

Comment:
Maybe make it clear that the error is caused by `torch._C.Value.type()`, so that readers know they are safe as long as the arg has the proper type?

Reply:
updated the docstring, still, didn't encourage users to feel safe with `from_name` and especially `from_dtype`